src/Database/Query.php

```diff
@@ -1452,6 +1458,27 @@ public function epilog(ExpressionInterface|string|null $expression = null)
         return $this;
     }
 
+    /**
+     * A string or expression that will be appended to the generated query as a comment
+     *
+     * ### Examples:
+     * ```
+     * $query->select('id')->where(['author_id' => 1])->comment('Filter for admin user');
+     * ```
+     *
+     * Comment content is raw SQL and not suitable for use with user supplied data.
+     *
+     * @param \Cake\Database\ExpressionInterface|string|null $expression The comment to be added
+     * @return $this
+     */
+    public function comment(ExpressionInterface|string|null $expression = null)
```

Comment:
Won't expressions generate placeholders that won't be part of the executed query? Would that break the query?

Reply:
I have added tests which use a query expression. Not really sure what you mean here.

Reply:
This accepts an arbitrary expression, but seems like just a string would suffice. 

Do we pass the ValueBinder to the comment expression which could allocate placeholders?

Reply:
I am fine with dropping the expression type since I don't see any value in having placeholders in comments.